src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -292,92 +196,186 @@ internal sealed class MeterProviderSdk : MeterProvider
         }
         else
         {
-            this.listener.InstrumentPublished = (instrument, listener) =>
-            {
-                bool enabledMeasurements = false;
+            // Everything double
+            this.listener.SetMeasurementEventCallback<double>(this.MeasurementRecordedDoubleSingleStream);
+            this.listener.SetMeasurementEventCallback<float>((instrument, value, tags, state) => this.MeasurementRecordedDoubleSingleStream(instrument, value, tags, state));
+
+            // Everything long
+            this.listener.SetMeasurementEventCallback<long>(this.MeasurementRecordedLongSingleStream);
+            this.listener.SetMeasurementEventCallback<int>((instrument, value, tags, state) => this.MeasurementRecordedLongSingleStream(instrument, value, tags, state));
+            this.listener.SetMeasurementEventCallback<short>((instrument, value, tags, state) => this.MeasurementRecordedLongSingleStream(instrument, value, tags, state));
+            this.listener.SetMeasurementEventCallback<byte>((instrument, value, tags, state) => this.MeasurementRecordedLongSingleStream(instrument, value, tags, state));
+
+            this.listener.MeasurementsCompleted = (instrument, state) => this.MeasurementsCompletedSingleStream(instrument, state);
+        }
+
+        this.listener.Start();
+
+        OpenTelemetrySdkEventSource.Log.MeterProviderSdkEvent("MeterProvider built successfully.");
+    }
+
+    internal Resource Resource { get; }
+
+    internal List<object> Instrumentations => this.instrumentations;
+
+    internal MetricReader? Reader => this.reader;
+
+    internal int ViewCount => this.viewConfigs.Count;
+
+    internal object? InstrumentPublished(Instrument instrument, bool listeningIsManagedExternally)
+    {
+        var listenToInstrumentUsingSdkConfiguration = this.shouldListenTo(instrument);
+
+        if (listeningIsManagedExternally && listenToInstrumentUsingSdkConfiguration)
+        {
+            OpenTelemetrySdkEventSource.Log.MetricInstrumentIgnored(
+                instrument.Name,
+                instrument.Meter.Name,
+                "Instrument belongs to a Meter which has been enabled externally and via a subscription on the provider. External subscription will be ignored in favor of the provider subscription.",
+                "Don't call AddMeter when also using external management.");
+            return null;
+        }
+        else if (!listeningIsManagedExternally && !listenToInstrumentUsingSdkConfiguration)
```

Comment:
nit: Switch the order of the operands. Most users would not be using the MetricsBuilder approach, so `!listeningIsManagedExternally` would always result in `true`.

Reply:
Updated